opm/output/eclipse/LinearisedOutputTable.hpp

```diff
@@ -0,0 +1,168 @@
+    class LinearisedOutputTable
+    {
+    public:
+        /// Constructor.
+        ///
+        /// \param[in] numTables Number of tables managed by internal
+        ///    buffer.  Typically corresponds to maximum value of a region
+        ///    ID vector such as SATNUM, IMBNUM, or PVTNUM.
+        ///
+        /// \param[in] numPrimary Number of primary look-up keys for the
+        ///    tabular data managed by the internal buffer.  Mostly relevant
+        ///    (i.e., greater than one) for miscible oil ("PVTO") or
+        ///    miscible gas ("PVTG") tables and typically corresponds to the
+        ///    number of Rs/Rv entries from the TABDIMS keyword.
+        ///
+        /// \param[in] numRows Number of rows in each sub-table
+        ///    corresponding to a single primary look-up key.  Typically the
+        ///    number of nodes (e.g., NSSFUN or NPPVT) of the corresponding
+        ///    input keyword.
+        ///
+        /// \param[in] numCols Number of columns in each sub-table (and main
+        ///    table).  Typically 5.
+        LinearisedOutputTable(const std::size_t numTables,
+                              const std::size_t numPrimary,
+                              const std::size_t numRows,
+                              const std::size_t numCols);
+
+        /// Retrieve iterator to start of \c numRows (contiguous) column
+        /// elements of a particular sub-table of a particular main table.
+        ///
+        /// \param[in] tableID Numeric ID of main table for which to
+        ///    retrieve a column.  Must be strictly less than \c numTables
+        ///    constructor argument.
+        ///
+        /// \param[in] primID Numeric ID of primary look-up key (sub-table)
+        ///    for which to retrieve a column.  Must be strictly less than
+        ///    \c numPrimary constructor argument.
+        ///
+        /// \param[in] colID Numeric ID of column to be retrieved from
+        ///    particular sub-table of particular main table.  Must be
+        ///    strictly less than \c numCols constructor argument.
+        ///
+        /// \return Iterator to start of contiguous column elements.
+        std::vector<double>::iterator
+        column(const std::size_t tableID,
+               const std::size_t primID,
+               const std::size_t colID);
+
+        /// Read-only access to internal data buffer.
+        ///
+        /// Mostly to support outputting all table data to external storage.
+        const std::vector<double>& getData() const;
+
+        /// Destructive access to internal data buffer.
+        ///
+        /// Mostly to support outputting all table data to external storage.
+        ///
+        /// \return \code std::move() \endcode of the internal data buffer.
+        std::vector<double> getDataDestructively();
+
+    private:
+        /// Internal buffer for tabular data.
+        std::vector<double> data_;
```

Comment:
I'm sorry to bother you with this - but please please; those trailing underscores hurt my eyes - there is no tradition for using them in the rest of opm-output.

Reply:
I can probably name it `m_*` if that's better, but I can't keep the existing name as there can't be two class-scope identifiers called `tabdims`.